codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -373,11 +398,11 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf buffer, List<Object> ou
          * read chunk, read and ignore the CRLF and repeat until 0
          */
         case READ_CHUNK_SIZE: try {
-            AppendableCharSequence line = lineParser.parse(buffer);
+            ByteBuf line = lineParser.parse(buffer);
             if (line == null) {
                 return;
             }
-            int chunkSize = getChunkSize(line.toString());
+            int chunkSize = getChunkSize(line.array(), line.readableBytes());
```

Comment:
this is not correct... you will need to pass in `line.arrayOffset() +  line.readerIndex()` as well as there is no guarantee that the array will start at offset 0.

Reply:
that's true, let me get there too, thanks!